image/image_test.go

```diff
@@ -0,0 +1,479 @@
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+			})
+		})
+	})
+
+	Convey("given clienter.Do returns 200 response", t, func() {
+		clienter := &dphttp.ClienterMock{
+			SetPathsWithNoRetriesFunc: func(paths []string) {
+				return
+			},
+			DoFunc: func(ctx context.Context, req *http.Request) (*http.Response, error) {
+				return &http.Response{
+					StatusCode: 200,
+				}, nil
+			},
+		}
+		clienter.SetPathsWithNoRetries([]string{path, "/healthcheck"})
+
+		imageClient := NewAPIClient(testHost)
+		imageClient.cli = clienter
+		check := initialState
+
+		Convey("when imageClient.Checker is called", func() {
+			err := imageClient.Checker(ctx, &check)
+			So(err, ShouldBeNil)
+
+			Convey("then the expected check is returned", func() {
+				So(check.Name(), ShouldEqual, service)
+				So(check.Status(), ShouldEqual, healthcheck.StatusOK)
+				So(check.StatusCode(), ShouldEqual, 200)
+				So(check.Message(), ShouldEqual, service+health.StatusMessage[healthcheck.StatusOK])
+				So(*check.LastChecked(), ShouldHappenAfter, timePriorHealthCheck)
+				So(*check.LastSuccess(), ShouldHappenAfter, timePriorHealthCheck)
+				So(check.LastFailure(), ShouldBeNil)
+			})
+
+			Convey("and client.Do should be called once with the expected parameters", func() {
+				doCalls := clienter.DoCalls()
+				So(doCalls, ShouldHaveLength, 1)
+				So(doCalls[0].Req.URL.Path, ShouldEqual, path)
+			})
+		})
+	})
+}
+
+func TestClient_GetImages(t *testing.T) {
+	Convey("given a 200 status is returned with an empty result list", t, func() {
+		searchResp, err := ioutil.ReadFile("./response_mocks/images_0.json")
+		So(err, ShouldBeNil)
+
+		mockdphttpCli := createHTTPClientMock(http.StatusOK, searchResp)
+		cli := Client{cli: mockdphttpCli, url: "http://localhost:8080"}
+
+		Convey("when GetImages is called", func() {
+			m, err := cli.GetImages(ctx)
+
+			Convey("a positive response is returned", func() {
+				So(err, ShouldBeNil)
+				So(m.Count, ShouldEqual, 0)
+				So(m.Items, ShouldBeEmpty)
+				So(m.Limit, ShouldEqual, 0)
+				So(m.Offset, ShouldEqual, 0)
+				So(m.TotalCount, ShouldEqual, 0)
+			})
+
+			Convey("and dphttpclient.Do is called 1 time", func() {
+				checkResponseBase(mockdphttpCli, http.MethodGet, "/images")
+			})
+		})
+	})
+
+	Convey("given a 200 status is returned with an single result list", t, func() {
+		searchResp, err := ioutil.ReadFile("./response_mocks/images_1.json")
+		So(err, ShouldBeNil)
+
+		mockdphttpCli := createHTTPClientMock(http.StatusOK, searchResp)
+		cli := Client{cli: mockdphttpCli, url: "http://localhost:8080"}
+
+		Convey("when GetImages is called", func() {
+			m, err := cli.GetImages(ctx)
+
+			Convey("a positive response is returned", func() {
+				So(err, ShouldBeNil)
+				So(m.Count, ShouldEqual, 1)
+				So(m.Items, ShouldNotBeEmpty)
+				So(m.Items, ShouldHaveLength, 1)
+				mItem := m.Items[0]
+				So(mItem.Id, ShouldResemble, "042e216a-7822-4fa0-a3d6-e3f5248ffc35")
+				So(mItem.Downloads, ShouldNotBeEmpty)
```

Comment:
NB. I've added a few checks on the object, but I'm not convinced that the unit test should be checking every field as that's really testing `json.Unmarshal` then rather than this library. I can add more if necessary but not sure there's much value.

Reply:
I agree that we would be testing the library, maybe it's not necessary to add more tests which would be actually testing the serialisation.